framework/test/gtest/BundleTrackerMethodTest.cpp

```diff
@@ -0,0 +1,480 @@
+  ASSERT_EQ(Bundle::State::STATE_ACTIVE, bundle.GetState())
+    << "Test bundle failed to start";
+  int postCount = bundleTracker.GetTrackingCount();
+
+  EXPECT_EQ(3, postCount - preCount)
+    << "Tracking count didn't increment by 3 after a bundle was modified 3 "
+       "times";
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest,
+       GetTrackingCountWorksAfterBundleRemovedByStateChange)
+{
+  auto stateMask =
+    BundleTracker<>::CreateStateMask(Bundle::State::STATE_INSTALLED);
+  BundleTracker<> bundleTracker(context, stateMask);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  ASSERT_EQ(Bundle::State::STATE_INSTALLED, bundle.GetState())
+    << "Test bundle failed to install";
+
+  int preCount = bundleTracker.GetTrackingCount();
+  bundle.Start(); // bundle: INSTALLED->resolved->starting->active
+  ASSERT_EQ(Bundle::State::STATE_ACTIVE, bundle.GetState())
+    << "Test bundle failed to start";
+  int postCount = bundleTracker.GetTrackingCount();
+
+  EXPECT_EQ(1, postCount - preCount)
+    << "Tracking count didn't increment by 1 after a bundle was removed";
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackingCountWorksAfterRemoveMethod)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  ASSERT_NO_THROW(bundleTracker.Open()) << "BundleTracker failed to start";
+  Bundle bundle = cppmicroservices::testing::InstallLib(
+    framework.GetBundleContext(), "TestBundleA");
+  ASSERT_EQ(Bundle::State::STATE_INSTALLED, bundle.GetState())
+    << "Test bundle failed to install";
+
+  // If Remove removes a bundle from being tracked, tracking count increments by 1
+  int preCount = bundleTracker.GetTrackingCount();
+  bundleTracker.Remove(bundle);
+  ASSERT_EQ(Bundle::State::STATE_INSTALLED, bundle.GetState())
+    << "Remove() altered the bundle being removed";
+  int postCount = bundleTracker.GetTrackingCount();
+
+  EXPECT_EQ(1, postCount - preCount)
+    << "Tracking count didn't increment by 1 after Remove() removed a bundle";
+
+  // If Remove doesn't remove a bundle from being tracked, tracking count stays the same
+  preCount = postCount;
+  bundleTracker.Remove(bundle);
+  postCount = bundleTracker.GetTrackingCount();
+
+  EXPECT_EQ(0, postCount - preCount)
+    << "Tracking count changed after Remove() didn't remove a bundle";
+  bundleTracker.Close();
+}
+
+TEST_F(BundleTrackerMethodTest, GetTrackingCountWorksWhenClosed)
+{
+  BundleTracker<> bundleTracker(context, all_states);
+  EXPECT_EQ(-1, bundleTracker.GetTrackingCount())
+    << "Tracking count of unopened BundleTracker was not -1";
+
+  bundleTracker.Open();
+  bundleTracker.Close();
+  
+  EXPECT_EQ(-1, bundleTracker.GetTrackingCount())
+    << "Tracking count of unopened BundleTracker was not -1";
+}
+
+TEST_F(BundleTrackerMethodTest, RemoveWorks)
```

Comment:
minor; rename testpoint name - remove "Works" to be more consistent with other test point names

Reply:
Done.